### ISSUE TITLE
Cloud CP/M disapearing programs, and files not being saved to disk.

### ISSUE THREAD

**Issue by SegamanXero** [DJSures/NABU-Internet-Adapter#3]
Internet Adapter version 2023.01.21.00 / Cloud CP/M 2.2 with BIOS v2.3b

Hoping this is the right place to report two issues I am seeing with Cloud CP/M and the file system, or at least make you aware you had not run into them yourself.

First issue seems to be as you use Cloud CP/M (I.E. doing DIR on A, switching to another drive doing DIR there, and then coming back to A for another DIR) seems to cause programs to disappear from the disk. Rebooting the NABU and reloading the boot disk fresh will restore the applications that disappeared prior.

The other issue I have been running into is with attempting to save documents. I will try to use ED to create a document and save on a blank disk. Running DIR shows the files on the disk, however reopening the file will result in a blank file with nothing in it despite previously writing and saving it. Looking at the disk image after the fact in a hex editor shows that the headers are being written for the files, but the contents are not being saved at all.

Both issues seem to happen regardless if the IA is pulling A,B, and G disk images from the cloud, or if I save them and store them locally with my local disk images.

Let me know if you have any questions.

Thanks again, you have done such amazing work.

**Comment by DJSures** [comment]
Hmm, maybe you were in a different user area when you changed drives. And when you return to the other drive, you're in a different user area. When changing user areas, it happens across all drives. 

The blank disk will need to have already had space allocated to it. The BIOS will not "append" to the size of a disk image. It expects the image file to be the correct size. For example, a NABURN image is 8mb, so you'll need that file to be 8mb.

The cloud images are not full-size padded because it would take too long to download. It's documented here: https://nabu.ca/cloud-cpm

![image](https://user-images.githubusercontent.com/49293683/213977944-42c0a07d-d18e-4199-a390-f0dcae82b6f5.png)


You will also see an error in the NABU Internet Adapter about writing outside of the file bounds...

![image](https://user-images.githubusercontent.com/49293683/213981962-97f52ae3-c027-4e6b-83ea-2e585284554e.png)

**Comment by DJSures** [comment]
I added an empty drive with 1mb allocated space you can download on the cloud/cpm page

![image](https://user-images.githubusercontent.com/49293683/213984219-203dc8fd-dc69-4d3d-8178-55601e60bec9.png)

**Comment by SegamanXero** [comment]
Thank you for your reply and information.

Nope was careful about being in the same user area as ED. I noticed you added the disk image padding step, as well as many other things in the documentation.

I noticed you did push some changes, and with 2.4b BIOS I can confirm that I am now able to save documents. Was not able to reopen them in ED, but I think that is a issue with me getting the hang of ED. I was able to confirm in a hex editor that content I saved with the e command for ED was on the disk image.

I also am not noticing the applications on A disappearing when swapping drives as well thus far.
